lambda-hot-reloading/typescript/README.md

```diff
@@ -0,0 +1,100 @@
+# Hot reloading for TypeScript Lambdas
+
+You can hot-reload your [TypeScript Lambda functions](https://docs.aws.amazon.com/lambda/latest/dg/lambda-typescript.html). We will check-out a simple example to create a simple `Hello World!` Lambda function using TypeScript.
+
+## Setting up the Lambda function
+
+Create a new Node.js project with `npm` or an alternative package manager:
+
+```sh
+$ npm init -y
+```
+
+Install the the [@types/aws-lambda](https://www.npmjs.com/package/@types/aws-lambda) and [esbuild](https://esbuild.github.io/) packages in your Node.js project:
+
+```sh
+$ npm install -D @types/aws-lambda esbuild
+```
+
+Create a new file named `index.ts`. Add the following code to the new file:
+
+```ts
+import { Context, APIGatewayProxyResult, APIGatewayEvent } from 'aws-lambda';
+export const handler = async (event: APIGatewayEvent, context: Context): Promise<APIGatewayProxyResult> => {
+  console.log(`Event: ${JSON.stringify(event, null, 2)}`);
+  console.log(`Context: ${JSON.stringify(context, null, 2)}`);
+  return {
+      statusCode: 200,
+      body: JSON.stringify({
+          message: 'Hello World!',
+      }),
+   };
+};
+```
+
+Add a build script to your `package.json` file:
+
+```json
+"scripts": {
+    "build": "esbuild index.ts --bundle --minify --sourcemap --platform=node --target=es2020 --outfile=dist/index.js --watch"
+}
+```
+
+The build script will use `esbuild` to bundle and minify the TypeScript code into a single JavaScript file, which will be placed in the `dist` folder.
```

Comment:
If --watch really works properly here, it might be good to amend that description to make it clear it will watch for changes.